src/main/java/com/google/api/codegen/transformer/java/JavaSchemaTypeNameConverter.java

```diff
@@ -92,32 +100,38 @@ public TypeName getTypeNameInImplicitPackage(String shortName) {
   }
 
   @Override
-  public TypeName getTypeName(Schema schema) {
+  public TypeName getTypeName(String key, Schema schema) {
     if (schema.type() == Type.ARRAY) {
       TypeName listTypeName = typeNameConverter.getTypeName("java.util.List");
-      TypeName elementTypeName = getTypeNameForElementType(schema, true);
+      TypeName elementTypeName = getTypeNameForElementType(null, schema, null, true);
```

Comment:
Done.
Simply passing along "key" (refactored to escapedSchemaName) everywhere instead of "null".